### ISSUE TITLE
Apostrophe in a url breaks clickable link creation

### ISSUE THREAD

**Issue by Gorrrg** [notepad-plus-plus/notepad-plus-plus#9031]
### Description of the Issue
When there's a url that contains an apostrophe, the parsing stops before the apostrophe which creates an incomplete clickable link.

### Steps to Reproduce the Issue
1. Paste a url into a text document, for example [https://en.wikipedia.org/wiki/Parkinson's_law](https://en.wikipedia.org/wiki/Parkinson's_law)
2. Click the partial link
3. You will end up at [https://en.wikipedia.org/wiki/Parkinson](https://en.wikipedia.org/wiki/Parkinson)

### Expected Behavior
The entire legitimate url should be incorporated into the clickable link

### Actual Behavior
Clickable link stops before the apostrophe.

### Debug Information
Notepad++ v7.9   (64-bit)
Build time : Sep 22 2020 - 03:19:04
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Pro (64-bit) 
OS Version : 2004
OS Build : 19041.572
Current ANSI codepage : 65001
Plugins : JsMapParser.NppPlugin.dll JSMinNPP.dll mimeTools.dll NppConverter.dll NppExport.dll NPPJSONViewer.dll XMLTools.dll

**Comment by Uhf7** [comment]
https://en.wikipedia.org/wiki/Murphy's_law ...

**Comment by Yaron10** [comment]
@Uhf7,

Thank you for improving the parser. 
Great work. :+1:

**Comment by dottoreCapitano** [comment]
In my python scripts, I sometimes have constructs like:
`  page = requests.get('https://www.tum.de/', headers=headers)`
Double-clicking the URL opens https://www.tum.de/' which gives an error in this case.
Do you have any idea how to realize a nice behaviour also in this situation?

**Comment by Uhf7** [comment]
@dottoreCapitano 
Thank you for the post. The current URL parser works without looking back, which means, it disregards what character is before the link. So, seemingly clear decisions like "the URL is enclosed in quotes or double quotes" are in many cases not respected as expected. I tried to look back before, but there where massive critics in #8921 and #8975. So I abandoned this. But I still think it's a reasonable way to improve the URL detection.

I'm going to revise the URL parser, but given the time for it I currently can invest, it will not happen before the end of the year. Modifying those things need a lot of time thinking about, I don't know how to say it, I mean,  in one piece, uninterrupted, and in the moment, I can only accumulate one or two consecutive hours per day to think about Notepad++. So, I'm a little bit limited here.
